tests/unit/test-extension-runners/test.extension-runners.js

```diff
@@ -536,28 +536,46 @@ describe('util/extension-runners', () => {
          const {extensionRunner, reloadStrategy} = prepare({
            stubExtensionRunner: {
              reloadAllExtensions: sinon.spy(
-               () => Promise.reject(Error('fake reload error'))
+               () => Promise.reject(new Error('fake reload error'))
              ),
            },
          });
 
          const fakeStdin = createFakeStdin();
          sinon.spy(fakeStdin, 'setRawMode');
 
+         // Stub the `fakeStdin.once` method to be able to wait
+         // once a promise resolved when the reloadStrategy method
+         // did call `stdin.once('keypress', ...)`.
+         const fakeStdinOnce = fakeStdin.once;
+         sinon.stub(fakeStdin, 'once');
+         fakeStdin.once.callsFake((...args) => {
+           if (args[0] === 'keypress') {
+             fakeStdin.emit('once-keypress-called');
```

Comment:
Rather than misusing the event emitter interface as a side effect, it might be more obvious to store the `resolve` function in a temporary variable, and call it from here.